From d769dd8f7c297c4443e9ab639b011dd2cedff339 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Thu, 2 Aug 2018 19:41:10 +0800
Subject: [PATCH 4298/5242] MLK-19117 ARM64: dts: imx8mm-evk: add usdhc2 cd
 support

commit  471eb5fb68c17f7eee07cd752bdfbd2d379be32d from
https://source.codeaurora.org/external/imx/linux-imx.git

imx8mm-evk revB board remove pwm led, and connect the gpio1_io15
to sd slot card detect pin. So this patch add usdhc2 cd feature
support, and remove the pwm led node in dts.

For gpio1_io15, need config to pull up mode, otherwise cd feature
will not work.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Reviewed-by: Dong Aisheng <Aisheng.dong@nxp.com>
(cherry picked from commit c24c9acf99b378130397980da2fe38e3cc411ed3)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts |   26 ++--------------------
 1 file changed, 2 insertions(+), 24 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
index 416175d..7268ee1 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm-evk.dts
@@ -45,17 +45,6 @@
 		#reset-cells = <0>;
 	};
 
-	pwmleds {
-		compatible = "pwm-leds";
-		ledpwm4 {
-			label = "PWM4";
-			pwms = <&pwm4 0 50000>;
-			max-brightness = <255>;
-			linux,default-trigger = "heartbeat";
-			active-low;
-		};
-	};
-
 	regulators {
 		compatible = "simple-bus";
 		#address-cells = <1>;
@@ -238,12 +227,6 @@
 			>;
 		};
 
-		pinctrl_pwm4: pwm4grp {
-			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO15_PWM4_OUT	0x16
-			>;
-		};
-
 		pinctrl_pmic: pmicirq {
 			fsl,pins = <
 				MX8MM_IOMUXC_GPIO1_IO03_GPIO1_IO3		0x41
@@ -387,6 +370,7 @@
 
 		pinctrl_usdhc2_gpio: usdhc2grpgpio {
 			fsl,pins = <
+				MX8MM_IOMUXC_GPIO1_IO15_GPIO1_IO15	0x1c4
 				MX8MM_IOMUXC_SD2_RESET_B_GPIO2_IO19	0x41
 			>;
 		};
@@ -898,12 +882,6 @@
 	status = "okay";
 };
 
-&pwm4 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm4>;
-	status = "okay";
-};
-
 &uart1 { /* BT */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1>;
@@ -959,8 +937,8 @@
 	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
+	cd-gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
 	bus-width = <4>;
-	non-removable;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
 };
-- 
1.7.9.5

